src/ripple/app/ledger/impl/LedgerReplayTask.cpp

```diff
@@ -0,0 +1,271 @@
+    , skipListAcquirer_(skipListAcquirer)
+{
+    JLOG(m_journal.trace()) << "Task ctor " << mHash;
+}
+
+LedgerReplayTask::~LedgerReplayTask()
+{
+    JLOG(m_journal.trace()) << "Task dtor " << mHash;
+}
+
+void
+LedgerReplayTask::init()
+{
+    JLOG(m_journal.debug()) << "Task start " << mHash;
+    ScopedLockType sl(mLock);
+    trigger(sl);
+    if (!isDone())
+        setTimer();
+}
+
+void
+LedgerReplayTask::trigger(ScopedLockType& sl)
+{
+    JLOG(m_journal.trace()) << "trigger " << mHash;
+    if (isDone() || !parameter_.full)
+        return;
+
+    if (!parent_)
+    {
+        parent_ = app_.getLedgerMaster().getLedgerByHash(parameter_.startHash);
+        if (!parent_)
+        {
+            parent_ = inboundLedgers_.acquire(
+                parameter_.startHash,
+                parameter_.startSeq,
+                InboundLedger::Reason::GENERIC);
+        }
+        if (parent_)
+        {
+            JLOG(m_journal.trace())
+                << "Got start ledger " << parameter_.startHash << " for task "
+                << mHash;
+        }
+    }
+
+    tryAdvance(sl);
+}
+
+void
+LedgerReplayTask::deltaReady(uint256 const& deltaHash)
+{
+    JLOG(m_journal.trace())
+        << "Delta " << deltaHash << " ready for task " << mHash;
+    ScopedLockType sl(mLock);
+    tryAdvance(sl);
+}
+
+void
+LedgerReplayTask::tryAdvance(ScopedLockType& sl)
+{
+    JLOG(m_journal.trace())
+        << "tryAdvance task " << mHash
+        << (parameter_.full ? ", full parameter"
+                            : ", waiting to fill parameter")
+        << ", deltaIndex=" << deltaToBuild << ", totalDeltas=" << deltas_.size()
+        << ", parent " << (parent_ ? parent_->info().hash : uint256());
+
+    bool shouldTry = !isDone() && parent_ && parameter_.full &&
+        parameter_.totalLedgers - 1 == deltas_.size();
+    if (!shouldTry)
+        return;
+
+    while (deltaToBuild < deltas_.size())
+    {
+        assert(parent_->seq() + 1 == deltas_[deltaToBuild]->ledgerSeq_);
+        if (auto l = deltas_[deltaToBuild]->tryBuild(parent_); l)
+        {
+            JLOG(m_journal.debug())
+                << "Task " << mHash << " got ledger " << l->info().hash
+                << " deltaIndex=" << deltaToBuild
+                << " totalDeltas=" << deltas_.size();
+            parent_ = l;
+            ++deltaToBuild;
+        }
+        else
+            break;
+    }
+
+    if (deltaToBuild >= deltas_.size())
+    {
+        mComplete = true;
+        JLOG(m_journal.info()) << "Completed " << mHash;
+    }
+}
+
+void
+LedgerReplayTask::updateSkipList(
+    uint256 const& hash,
+    std::uint32_t seq,
+    std::vector<ripple::uint256> const& sList)
+{
+    {
+        ScopedLockType sl(mLock);
+        if (!parameter_.update(hash, seq, sList))
+        {
+            JLOG(m_journal.error()) << "Parameter update failed " << mHash;
+            mFailed = true;
+            return;
+        }
+    }
+
+    replayer_.createDeltas(shared_from_this());
```

Comment:
Perhaps `this->trigger(sl)` should be called here to start fetching the start ledger if need be.

Reply:
You are right! Added.